RecoParticleFlow/PFClusterProducer/interface/PFEcalEndcapRecHitCreator.h

```diff
@@ -0,0 +1,142 @@
+#ifndef RecoParticleFlow_PFClusterProducer_PFEcalEndcapRecHitCreator_h
+#define RecoParticleFlow_PFClusterProducer_PFEcalEndcapRecHitCreator_h
+
+#include "RecoParticleFlow/PFClusterProducer/interface/PFRecHitCreatorBase.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
+
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+
+
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/CaloGeometry/interface/TruncatedPyramid.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "Geometry/EcalAlgo/interface/EcalEndcapGeometry.h"
+#include "Geometry/EcalAlgo/interface/EcalBarrelGeometry.h"
+#include "Geometry/CaloTopology/interface/EcalEndcapTopology.h"
+#include "Geometry/CaloTopology/interface/EcalBarrelTopology.h"
+#include "Geometry/CaloTopology/interface/EcalPreshowerTopology.h"
+#include "RecoCaloTools/Navigation/interface/CaloNavigator.h"
+
+class PFEcalEndcapRecHitCreator :  public  PFRecHitCreatorBase {
+
+ public:  
+  PFEcalEndcapRecHitCreator(const edm::ParameterSet& iConfig,edm::ConsumesCollector& iC):
+    PFRecHitCreatorBase(iConfig,iC)
+    {
+      recHitToken_ = iC.consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("src"));
+      srFlagToken_ = iC.consumes<EESrFlagCollection>(iConfig.getParameter<edm::InputTag>("srFlags"));
+      elecMap_ = 0;
+    }
+
+    void importRecHits(std::unique_ptr<reco::PFRecHitCollection>&out,std::unique_ptr<reco::PFRecHitCollection>& cleaned ,const edm::Event& iEvent,const edm::EventSetup& iSetup) {
+
+      beginEvent(iEvent,iSetup);
+ 
+      edm::Handle<EcalRecHitCollection> recHitHandle;
+
+      edm::ESHandle<CaloGeometry> geoHandle;
+      iSetup.get<CaloGeometryRecord>().get(geoHandle);
+  
+      iEvent.getByToken(srFlagToken_,srFlagHandle_);
+
+      // get the ecal geometry
+      const CaloSubdetectorGeometry *gTmp = 
+	geoHandle->getSubdetectorGeometry(DetId::Ecal, EcalEndcap);
+
+      const EcalEndcapGeometry *ecalGeo =dynamic_cast< const EcalEndcapGeometry* > (gTmp);
+
+      iEvent.getByToken(recHitToken_,recHitHandle);
+      for(const auto& erh : *recHitHandle ) {      
+	const DetId& detid = erh.detid();
+	auto energy = erh.energy();
+	auto time = erh.time();
+
+        bool hi = isHighInterest(detid);
+        
+	const CaloCellGeometry * thisCell= ecalGeo->getGeometry(detid);
+  
+	// find rechit geometry
+	if(!thisCell) {
+	  edm::LogError("PFEcalEndcapRecHitCreator")
```

Comment:
To be clear, my point was to have an Exception here instead of skipping the hit.